pymoskito/simulation_gui.py

```diff
@@ -1503,16 +1499,25 @@ def find_all_plot_docks(self):
 
         return list
 
-    def export_csv(self, plot_item, name, *args):
-        exporter = pg.exporters.CSVExporter(plot_item)
-        filename = QFileDialog.getSaveFileName(self,
-                                               "CSV export", name + ".csv",
-                                               "CSV Data (*.csv)")
-        if not filename[0]:
-            return
+    def exportPlotItem(self, plot_item, name, coord_item, time_item):
+        dataPoints = {}
+        for i, c in enumerate(plot_item.curves):
+            if c.getData() is None:
+                continue
+            if len(c.getData()) > 2:
+                self._logger.warning('Can not handle the amount of data!')
+                continue
+            dataPoints[c.name() + str('.0')] = c.getData()[0]
+            dataPoints[c.name() + str('.1')] = c.getData()[1]
 
-        exporter.export(filename[0])
-        self._logger.info("Plot '{}' saved as '{}'".format(name, filename[0]))
+        self.export_csv(dataPoints)
+
+    def export_csv(self, dataPoints):
+        exporter = CSVExporter(dataPoints)
```

Comment:
It seems like yo do not favour the pyqtgraph csv export function. Instead of reinventing the wheel here, maybe we could extended the `CSVExporter` to an `Exporter` class which offers a copy constructor that fills an internal pandas data-frame with plot data and then exposes its various export functionalities?